Mathlib/Data/Fintype/Basic.lean

```diff
@@ -443,6 +443,13 @@ protected def subtype {p : α → Prop} (s : Finset α) (H : ∀ x : α, x ∈ s
 def ofFinset {p : Set α} (s : Finset α) (H : ∀ x, x ∈ s ↔ x ∈ p) : Fintype p :=
   Fintype.subtype s H
 
+/-- Given a fintype `α`, a subtype of `α` is also a fintype. -/
+def subtypeOfFintype [Fintype α] (P : α → Prop) [DecidablePred P] :
+    Fintype {v // P v} where
+  elems := attachWith (univ.filter P) P fun _ h => (mem_filter.mp h).2
+  complete := fun ⟨x, px⟩ => Multiset.mem_pmap.mpr
+    ⟨x, Multiset.mem_filter.mpr ⟨mem_univ _, px⟩, rfl⟩
+
```

Comment:
```suggestion
/-- Given a fintype `α`, a subtype of `α` is also a fintype. -/
def subtypeOfFintype [Fintype α] (P : α → Prop) [DecidablePred P] : Fintype {v // P v} :=
  Fintype.subtype (univ.filter P) (by simp)
```
instead of making two new definitions, you should instead build this out of existing ones, and avoid making attachWith.